File: 백준/Silver/10828. 스택/스택.py
import sys
input = sys.stdin.readline
n = int(input())
stack = []
for _ in range(n):
  cmd = sys.stdin.readline().split()
  what = cmd[0]
  if what == 'push':
    stack.append(cmd[1])
  elif what == 'pop':
    if stack:
      print(stack.pop())
    else:
      print(-1)
  elif what == 'size':
      print(len(stack))
  elif what == 'empty':
    print(0 if stack else 1)
  elif what == 'top':
    if stack:
      print(stack[-1])
    else:
      print(-1)
  

File: 백준/Silver/10828. 스택/스택.js
class Stack {
  constructor() {
    this.stack = [];
  }

  push(newElement) {
    this.stack.push(newElement);
  }

  pop() {
    if (this.empty()) return -1;
    return this.stack.pop();
  }

  size() {
    return this.stack.length;
  }

  empty() {
    return this.stack.length === 0 ? 1 : 0;
  }

  top() {
    if (this.empty()) return -1;
    return this.stack.at(-1);
  }
}

const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const [n, ...arr] = fs.readFileSync(filePath).toString().trim().split("\n");
const commands = arr.map((command) => command.split(" "));

const stack = new Stack();
let answer = [];

for (let [com, num] of commands) {
  if (com === "push") {
    stack.push(num);
  } else {
    answer.push(stack[com]());
  }
}

console.log(answer.join("\n"));
